### PR TITLE
Feature/temperature: Ambient temperature pipeline

### PR DESCRIPTION
**Pull request by rararani**

### CODE REVIEW

File: payload_designer/pipelines/pipe_ambient_temp.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        operating_ranges.append(comp.calculate_operating_temp_range())
+
+    min_range = min(operating_ranges)
+    min_component = filtered_components[operating_ranges.index(min_range)]
+
+    print(f"Component: {min_component}")
+    print(
+        f"range: {min_component.min_operating_temp} - \
+        {min_component.max_operating_temp}"
+    )
```
Comment (DM1122) [inline payload_designer/pipelines/pipe_ambient_temp.py:34]: This algorithm assumes all the temperature ranges are overlapping, which might not be the case. Check out the attached document which shows some possible range cases. There's some logic that's needed to determine what this function should do in each of them. This reminds me of a typical interview technical question.

[FINCH Operating Temperature Ranges by Component Cases.pdf](https://github.com/spacesys-finch/payload-designer/files/8208902/FINCH.Operating.Temperature.Ranges.by.Component.Cases.pdf)

**Review by DM1122 — CHANGES_REQUESTED** [review]
Solid progress! Great idea to create a base component class to house the operating temp ranges. Was going to get around to it eventually, glad you got it done. Later on, mass and volume envelopes will be stored as attributes in the base class to reduce redundancy.

File: payload_designer/components/lenses.py [hunk 2]
```diff
@@ -3,18 +3,20 @@
 # stdlib
 import logging
 import math
+from lib2to3.pytree import Base
```
[the remark this answers is not in the document]
Comment (rararani) [reply payload_designer/components/lenses.py:6]: Sometimes VSCode auto imports randoms stuff. It must've saw "Base" and decided to auto import this library lol, I'll remove it

File: payload_designer/pipelines/pipe_ambient_temp.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+"""Calculate the net operating temperature across all components."""
+# project
+from payload_designer.components.components import Components
+
+
+def get_components():
+    obj = Components()
+    return [
+        a for a in dir(obj) if not a.startswith("__") and not callable(getattr(obj, a))
+    ]
```
[the remark this answers is not in the document]
Comment (rararani) [reply payload_designer/pipelines/pipe_ambient_temp.py:10]: @DM1122 what is the focuser here? It's defined the same way as the collimator.

[the remark this answers is not in the document]
Comment (DM1122) [reply payload_designer/pipelines/pipe_ambient_temp.py:10]: The focuser is the exact same lens component as the the collimator. They're just two of the same lenses in different parts of the track. The collimator makes the incident light parallel to itself, the collimator focuses the parallel light onto the sensor face.

**Review by DM1122 — APPROVED** [review]

**Pull request merged**